tasks/eva_3667/src/main/groovy/eva3667/remediate_lowercase_nucleotide.groovy

```diff
@@ -0,0 +1,362 @@
+package eva3667
+
+import com.mongodb.BasicDBObject
+import com.mongodb.client.MongoCollection
+import com.mongodb.client.model.Filters
+import com.mongodb.client.model.Updates
+import groovy.cli.picocli.CliBuilder
+import org.bson.Document
+import org.bson.conversions.Bson
+import org.slf4j.Logger
+import org.slf4j.LoggerFactory
+import org.springframework.beans.factory.annotation.Autowired
+import org.springframework.boot.CommandLineRunner
+import org.springframework.boot.autoconfigure.SpringBootApplication
+import org.springframework.boot.builder.SpringApplicationBuilder
+import org.springframework.data.mongodb.core.MongoTemplate
+import org.springframework.data.mongodb.core.convert.DefaultMongoTypeMapper
+import org.springframework.data.mongodb.core.convert.MappingMongoConverter
+import org.springframework.data.mongodb.core.query.Criteria
+import org.springframework.data.mongodb.core.query.Query
+import org.springframework.data.util.Pair
+import uk.ac.ebi.eva.commons.models.data.Variant
+import uk.ac.ebi.eva.commons.models.data.VariantSourceEntity
+import uk.ac.ebi.eva.commons.models.mongo.entity.VariantDocument
+import uk.ac.ebi.eva.commons.models.mongo.entity.subdocuments.VariantSourceEntryMongo
+import uk.ac.ebi.eva.commons.models.mongo.entity.subdocuments.VariantStatsMongo
+import uk.ac.ebi.eva.commons.mongodb.entities.subdocuments.HgvsMongo
+
+import java.nio.file.Paths
+import java.util.stream.Collectors
+
+import static org.springframework.data.mongodb.core.query.Criteria.where
+
+def cli = new CliBuilder()
+cli.workingDir(args: 1, "Path to the working directory where processing files will be kept", required: true)
+cli.envPropertiesFile(args: 1, "Properties file with db details to use for remediation", required: true)
+cli.dbName(args: 1, "Database name that needs to be remediated", required: true)
+def options = cli.parse(args)
+if (!options) {
+    cli.usage()
+    System.exit(1)
+}
+
+
+// Run the remediation application
+new SpringApplicationBuilder(RemediationApplication.class).properties([
+        'spring.config.location'      : options.envPropertiesFile,
+        'spring.data.mongodb.database': options.dbName])
+        .run(options.workingDir, options.dbName)
+
+
+@SpringBootApplication
+class RemediationApplication implements CommandLineRunner {
+    private static Logger logger = LoggerFactory.getLogger(RemediationApplication.class)
+
+    public static final String VARIANTS_COLLECTION = "variants_2_0"
+    public static final String FILES_COLLECTION = "files_2_0"
+    public static final String ANNOTATIONS_COLLECTION = "annotations_2_0"
+    public static final String FILES_COLLECTION_STUDY_ID_KEY = "sid"
+    public static final String FILES_COLLECTION_FILE_ID_KEY = "fid"
+
+    public static final String REGEX_PATTERN = '(?:\\S+_){2}([ACGTN]*[^ACGTN_]+[ACGTN]*)?_([ACGTN]*[^ACGTN_]+[ACGTN]*)?(?=\\s*$)'
```

Comment:
The given regex does not work for correctly for cases where chromosome name has underscore. It might catche cases where ref and alt does not have any lower case but they will get filtered out in 2nd check.
Based on the discussion leaving it as it is, as we don't expect these to be large in number.